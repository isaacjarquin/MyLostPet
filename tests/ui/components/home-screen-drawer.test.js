import "react-native"
import React from "react"
import HomeScreenDrawer from "../../../components/home-screen-drawer"

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import renderer from "react-test-renderer"

describe("HomeScreenDrawer", () => {
    const navigate = jest.fn()

    const props = {
        navigation: {
            navigate: navigate
        }
    };

    beforeEach(() => {
        wrapper = shallow(<HomeScreenDrawer {...props} />);
    });

    afterEach(() => {
        navigate.mockReset()
    });

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    describe("_whoWeAre", () => {
        beforeEach(() => {
            wrapper.instance()._whoWeAre()
        })

        it("navigate to WhoWeAre", () => {
            expect(navigate).toHaveBeenCalledTimes(1);
            expect(navigate).toHaveBeenCalledWith("WhoWeAre")
        })
    })

    describe("_termsAndConditions", () => {
        beforeEach(() => {
            wrapper.instance()._termsAndConditions()
        })

        it("navigate to WhoWeAre", () => {
            expect(navigate).toHaveBeenCalledTimes(1);
            expect(navigate).toHaveBeenCalledWith("TermsAndConditions")
        })
    })

    describe("_howToUse", () => {
        beforeEach(() => {
            wrapper.instance()._howToUse()
        })

        it("navigate to WhoWeAre", () => {
            expect(navigate).toHaveBeenCalledTimes(1);
            expect(navigate).toHaveBeenCalledWith("HowToUse")
        })
    })
})
