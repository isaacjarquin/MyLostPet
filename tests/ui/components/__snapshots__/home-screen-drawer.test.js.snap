// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreenDrawer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../assets/images/xira.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
          "borderColor": "#d6d7da",
          "borderWidth": 0.5,
          "opacity": 0.6,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Quienes somos
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
          "borderColor": "#d6d7da",
          "borderWidth": 0.5,
          "opacity": 0.6,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Como usar la aplicación
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "backgroundColor": "grey",
          "borderColor": "#d6d7da",
          "borderWidth": 0.5,
          "opacity": 0.6,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Contacto
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
          "borderColor": "#d6d7da",
          "borderWidth": 0.5,
          "opacity": 0.6,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Términos y condiciones
      </Text>
    </TouchableOpacity>
  </ImageBackground>
</View>
`;
